### PR TITLE
Added array value

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: multiversx_sdk/abi/array_value_test.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import pytest
+
+from multiversx_sdk.abi.array_value import ArrayValue
+from multiversx_sdk.abi.biguint_value import BigUIntValue
+from multiversx_sdk.abi.fields import Field
+from multiversx_sdk.abi.small_int_values import U32Value
+from multiversx_sdk.abi.struct_value import StructValue
+
+
```
Comment (andreibancioiu) [inline multiversx_sdk/abi/array_value_test.py:11]: For the new `ArrayValue` class, let's add some unit tests for encoding / decoding, as well, similar to the ones in Go:

https://github.com/multiversx/mx-sdk-abi-go/blob/main/abi/listValue_test.go#L10

File: multiversx_sdk/abi/array_value.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        self.length = length
+        self.items = items or []
+
+        if len(self.items):
+            self.guard_length(self.items)
+
+        self.item_creator = item_creator
+
+    def encode_nested(self, writer: io.BytesIO):
+        encode_length(writer, len(self.items))
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/abi/array_value.py:23]: Indeed, forgot to remove it.

File: multiversx_sdk/abi/array_value.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self._encode_list_items(writer)
+
+    def encode_top_level(self, writer: io.BytesIO):
+        self._encode_list_items(writer)
+
+    def decode_nested(self, reader: io.BytesIO):
+        length = decode_length(reader)
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/abi/array_value.py:30]: Indeed, removed it.

File: multiversx_sdk/abi/array_value_test.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk/abi/array_value_test.py:11]: Added.

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk/abi/array_value_test.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    encoded = codec.encode_nested(array_value)
+    assert encoded.hex() == "000100020003"
+
+
+def test_decode_top_level():
+    codec = Codec()
+    data = bytes.fromhex("000100020003")
+
+    destinattion = ArrayValue(length=3, item_creator=lambda: U16Value())
```
Comment (axenteoctavian) [inline multiversx_sdk/abi/array_value_test.py:95]: little typo for `destination`

File: multiversx_sdk/abi/array_value_test.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    codec.decode_top_level(data, destinattion)
+
+    assert destinattion.items == [U16Value(1), U16Value(2), U16Value(3)]
+
+
+def test_decode_nested():
+    codec = Codec()
+    data = bytes.fromhex("000100020003")
+
+    destinattion = ArrayValue(length=3, item_creator=lambda: U16Value())
```
Comment (axenteoctavian) [inline multiversx_sdk/abi/array_value_test.py:105]: same typo here

File: multiversx_sdk/abi/array_value_test.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk/abi/array_value_test.py]: fixed

File: multiversx_sdk/abi/array_value_test.py (hunk 5 above)
Comment (popenta) [reply multiversx_sdk/abi/array_value_test.py]: fixed

**Review by andreibancioiu — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**